### ISSUE TITLE
Remember to check TLS 1.2

### ISSUE THREAD

**Issue by fabrouy** [ruckus/quickbooks-ruby#441]
Just a friendly reminder:

https://developer.intuit.com/hub/blog/2018/07/11/tls-1-0-1-1-deprecation-intuit-developers 

How to test if your application is connecting to TLS 1.2 URL successfully after the required changes:
Make an HTTP call to this URL from your code https://tlstest.intuit.com.
It should give you a ‘success’ response.
Or

Make an HTTP call to this URL from your code https://tlstest.intuit.com/1_2.json
It should give you the following json response-

{
“status”: 200,
“message”: “Success!”
}

**Comment by fabrouy** [comment]
Please close this issue after tomorrow, it's deadline.

**Comment by drewish** [comment]
@fabrouy you should be able to close issues you created

**Comment by nmichels** [comment]
@fabrouy @drewish I can successfully request the url `https://tlstest.intuit.com/1_2.json` with Net::HTTP from my code, but still I've had to disable temporarily QBO integration because of the following error: ` Errno::ECONNRESET: Connection reset by peer - SSL_connect`
Any advice on that? Thanks in advance.
PS.: I'm using OAuth2 and consequently pointing to this branch 389-oauth2

**Comment by ruckus** [comment]
@nmichels can you post both versions of your TLSTest code?

**Comment by nmichels** [comment]
@ruckus I don't know if I'm missing something, but I did run the following from heroku console:
```
uri = URI.parse("https://tlstest.intuit.com/1_2.json")
Net::HTTP.get(uri)
"{\n    \"status\": 200,\n    \"message\": \"Success!\"\n}\n"
```

Net::HTTP returns the expected result. Thanks!

**Comment by nmichels** [comment]
I know that OAuth2 uses faraday behind the scenes but I couldn't figure out the easiest way to test it, can I use something similar to the `webapp/app.rb` script and run the block `get '/oauth2callback' do` to check what's going on?

**Comment by ruckus** [comment]
> can I use something similar to the webapp/app.rb script and run the block get '/oauth2callback' do to check what's going on?

yeah thats one way. You could also maybe get ahold of a console and somehow try to call `base_service#do_http_raw_get` using `:send` and what not.

But yeah `webapp/app.rb` would probably be faster to get going

**Comment by nmichels** [comment]
That's pretty much weird, I've tested the integration in production (I've just hidden it from the final user) and no SSL errors happened today.
I think it was a possible service unavailability last week and not TLS 1.0 and 1.1that caused the errors. I'll confirm that with Intuit Support.
